subsys/bluetooth/host/gatt.c

```diff
@@ -4124,7 +4149,27 @@ static int ccc_set(const char *name, size_t len_rd, settings_read_cb read_cb,
 	return 0;
 }
 
+#if defined(CONFIG_BT_SETTINGS_CCC_LOAD_ON_DEMAND)
+static int ccc_set_direct(const char *key, size_t len, settings_read_cb read_cb,
+			  void *cb_arg, void *param)
+{
+	const char *name;
+
+	BT_DBG("key: %s", log_strdup((const char *)param));
+
+	/* Only "bt/ccc" settings should ever come here */
```

Comment:
Do we really need to check it? Maybe assertion would be better - if we get other settings name it seems like an internal code error, isn't it?

Reply:
More than a just a check, this is mostly meant to extract the const char *name that is needed by the ccc_set function (it is meant to avoid re-implementing what ccc_set already does in ccc_set_direct).

Reply:
Right. Ok then.